File: darkwater-common/src/main/java/kiinse/plugins/darkwaterapi/common/initialize/RegisterCommands.java
package kiinse.plugins.darkwaterapi.common.initialize;

import kiinse.plugins.darkwaterapi.api.DarkWaterJavaPlugin;
import kiinse.plugins.darkwaterapi.api.exceptions.CommandException;
import kiinse.plugins.darkwaterapi.common.commands.darkwaterapi.DarkWaterCommands;
import kiinse.plugins.darkwaterapi.common.commands.darkwaterapi.DarkWaterTab;
import kiinse.plugins.darkwaterapi.common.commands.locale.LocaleCommands;
import kiinse.plugins.darkwaterapi.common.commands.locale.LocaleTab;
import kiinse.plugins.darkwaterapi.common.commands.statistic.StatisticCommands;
import kiinse.plugins.darkwaterapi.core.commands.CommandManager;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class RegisterCommands {

    public RegisterCommands(@NotNull DarkWaterJavaPlugin plugin) throws NullPointerException, CommandException {
        plugin.sendLog("Registering commands...");
        new CommandManager(plugin)
                .registerCommand(new LocaleCommands(plugin))
                .registerCommand(new DarkWaterCommands(plugin))
                .registerCommand(new StatisticCommands(plugin));
        Objects.requireNonNull(plugin.getCommand("locale")).setTabCompleter(new LocaleTab(plugin));
        Objects.requireNonNull(plugin.getCommand("darkwater")).setTabCompleter(new DarkWaterTab(plugin));
        plugin.sendLog("Commands registered");
    }
}

File: common/src/main/kotlin/kiinse/me/plugins/darkwaterapi/common/initialize/RegisterCommands.kt
package kiinse.me.plugins.darkwaterapi.common.initialize

import kiinse.me.plugins.darkwaterapi.api.DarkWaterJavaPlugin
import kiinse.me.plugins.darkwaterapi.common.commands.darkwaterapi.DarkWaterCommands
import kiinse.me.plugins.darkwaterapi.common.commands.darkwaterapi.DarkWaterTab
import kiinse.me.plugins.darkwaterapi.common.commands.locale.LocaleCommands
import kiinse.me.plugins.darkwaterapi.common.commands.locale.LocaleTab
import kiinse.me.plugins.darkwaterapi.common.commands.statistic.StatisticCommands
import kiinse.me.plugins.darkwaterapi.core.commands.CommandManager

class RegisterCommands(plugin: DarkWaterJavaPlugin) {
    init {
        plugin.sendLog("Registering commands...")
        CommandManager(plugin)
                .registerCommand(LocaleCommands(plugin))
                .registerCommand(DarkWaterCommands(plugin))
                .registerCommand(StatisticCommands(plugin))
        plugin.getCommand("locale")!!.tabCompleter = LocaleTab(plugin)
        plugin.getCommand("darkwater")!!.tabCompleter = DarkWaterTab(plugin)
        plugin.sendLog("Commands registered")
    }
}